### ISSUE TITLE
Issue with cupy.random.XORWOW when generating random numbers beyond 2.1 billion

### ISSUE THREAD

**Issue by tairaskar** [cupy/cupy#8241]
### Description

I'm currently working with CuPy and have encountered an issue with the cupy.random.XORWOW generator when attempting to generate random numbers beyond 2.1 billion.

To provide some context, the generator works perfectly for generating random numbers up to 2.1 billion. However, when I try to generate more than that, it throws an error.: 

```
File "cupy/_core/core.pyx", line 1745, in cupy._core.core._ndarray_base.__str__
  File "cupy/_core/core.pyx", line 1856, in cupy._core.core._ndarray_base.get
  File "cupy/cuda/memory.pyx", line 561, in cupy.cuda.memory.MemoryPointer.copy_to_host
  File "cupy_backends/cuda/api/runtime.pyx", line 581, in cupy_backends.cuda.api.runtime.memcpy
  File "cupy_backends/cuda/api/runtime.pyx", line 144, in cupy_backends.cuda.api.runtime.check_status
cupy_backends.cuda.api.runtime.CUDARuntimeError: cudaErrorIllegalAddress: an illegal memory access was encountered
```

I've verified that there is plenty of GPU memory available, so it doesn't seem to be a memory limitation. I'm reaching out to seek advice and inquire if there might be a trivial fix or workaround for this issue. If anyone has encountered a similar situation or has insights into potential solutions, I would greatly appreciate your guidance.

### To Reproduce

```
import cupy as cp

array_size = 2200000000

gen = cp.random.Generator(cp.random.XORWOW(1234))
random_array = gen.random(array_size, dtype=cp.float32)
```

### Installation

None

### Environment

```
# Paste the output here
```

**Comment by tairaskar** [comment]
it seems we have to pass long int to cupy.random.XORWOW. 

array_size = cp.int64(2200000000) does not work.

**Comment by kmaehashi** [comment]
Thanks for reporting @tairaskar. This should be fixed in #8243.
